src/components/MainVideoStream.vue

```diff
@@ -1,45 +1,70 @@
-        <v-responsive aspect-ratio="16 / 9" class="border">
-            <video
-                id="mainVideo"
-                autoplay
-                class="video"
-                ref="mainVideo"
-            ></video>
+        <v-responsive id="stream" ref="stream" class="capture-area-selection">
+            <video id="mainVideo" class="video" autoplay></video>
+            <VueDragResize
+                v-if="rerender"
+                v-for="captureArea in captureAreas"
+                :key="captureArea.getId()"
+                :w="captureArea.getWidth()"
+                :h="captureArea.getHeight()"
+                :x="captureArea.getLeft()"
+                :y="captureArea.getTop()"
+                :parentW="wParent"
+                :parentH="hParent"
+                :parentScaleX="1"
+                :parentScaleY="1"
+                :isActive="false"
+                :snapToGrid="false"
+                :aspectRatio="false"
+                :preventActiveBehavior="false"
+                :parentLimitation="true"
+                @resizing="changeSize($event, captureArea)"
+                @dragging="changePosition($event, captureArea)"
+                class="background"
+                ref="drag"
+            >
+                <p>
+                    Postioning {{ captureArea.getTop() }} x
+                    {{ captureArea.getLeft() }}
+                </p>
+                <p>
+                    Capture Area Properties {{ captureArea.getWidth() }} x
+                    {{ captureArea.getHeight() }}
+                </p>
+            </VueDragResize>
         </v-responsive>
-        <div>
-            x: {{ x }} y: {{ y }} Is Outside: {{ isOutside }}
-
-            Height: {{ height }} Width: {{ width }}
-        </div>
     </div>
 </template>
 
 <script setup lang="ts">
-import { ref, onMounted } from 'vue';
-import { useMouseInElement, useElementSize } from '@vueuse/core';
+import { ref, onMounted, reactive } from 'vue';
+import { useElementSize } from '@vueuse/core';
 import { Vigad } from '@/proc/Vigad';
+// @ts-ignore
+import VueDragResize from 'vue3-drag-resize';
+import { rerender } from './Rerender';
 
-// Get singelton instance reference to vigad
+/**
+ * Get singelton instance reference to vigad
+ */
 const vigad = Vigad.getInstance();
 
-// Get singelton instance reference to streamHandler
+/**
+ * Get singelton instance reference to streamHandler
+ */
 const streamHandler = vigad.getStreamHandlerInstance();
 
-// Mouse position relative to the element and is in and out
-const mainVideo = ref<HTMLVideoElement | null>(null);
-
-// Mouse coordinates relative to the element and if cursor is inside of the element
-const { x, y, isOutside } = useMouseInElement(mainVideo);
-
-// Element size
-const { width, height } = useElementSize(mainVideo);
+/**
+ * Get reference to all the capture areas that are currently active
+ */
+const captureAreas = reactive(vigad.getAllCaptureAreas());
```

Comment:
Is it possible to import the entire vigad-class as reactive instead of only the capture areas?

Reply:
Yes it is possible, that's probably a good idea